packages/legend-data-cube/src/stores/core/DataCubeEngine.tsx

```diff
@@ -235,6 +236,18 @@ export abstract class DataCubeEngine {
     source: DataCubeSource,
   ): V1_AppliedFunction | undefined;
 
+  // ---------------------------------- CACHING --------------------------------------
+
+  async initializeCache(
+    source: DataCubeSource,
+  ): Promise<CachedDataCubeSource | void> {
```

Comment:
avoid return mixed type with `void`, it's hard to handle downstream
```suggestion
  ): Promise<CachedDataCubeSource | undefined> {
    return undefined;
```